src/bmpcodec.c

```diff
@@ -827,6 +839,34 @@ gdip_read_BITMAPINFOHEADER (void *pointer, ImageSource source, BITMAPV5HEADER *b
 	}
 
     return Ok;
+	
+fill_rgb_masks:
+	if (bmi->bV5BitCount != 16) {
+		bmi->bV5RedMask = 0;
+		bmi->bV5GreenMask = 0;
+		bmi->bV5BlueMask = 0;
+		return Ok;
+	}
+
+	dw = 0;
+	size_read = gdip_read_bmp_data (pointer, data_read, size, source);
+	if (size_read < size)
+		return OutOfMemory;
+	bmi->bV5RedMask = (data_read[3]<<24 | data_read[2]<<16 | data_read[1]<<8 | data_read[0]);
+
+	dw = 0;
+	size_read = gdip_read_bmp_data (pointer, data_read, size, source);
+	if (size_read < size)
+		return OutOfMemory;
+	bmi->bV5GreenMask = (data_read[3]<<24 | data_read[2]<<16 | data_read[1]<<8 | data_read[0]);
+
+	dw = 0;
+	size_read = gdip_read_bmp_data (pointer, data_read, size, source);
+	if (size_read < size)
+		return OutOfMemory;
+	bmi->bV5BlueMask = (data_read[3]<<24 | data_read[2]<<16 | data_read[1]<<8 | data_read[0]);
+
+	return Ok;
 }
```

Comment:
This doesn't sound right. I don't think the masks are even present there with BITMAPINFOHEADER when `biCompression` is `BI_RGB`. In that case it should use the default 16 bits per pixel values:

```
bmi->red_mask = 0x7C00;
bmi->green_mask = 0x3E0;
bmi->blue_mask = 0x1F;
```

Reply:
The cause of this issue was https://github.com/mono/libgdiplus/issues/285 which needs to be fixed (I'll do so separately)

Addressed this feedback though